src/gpb_compile.erl

```diff
@@ -3958,11 +3961,15 @@ format_hrl(Mod, Defs, Opts) ->
        ?f("-endif.~n")]).
 
 format_msg_record(Msg, Fields, Opts, Defs) ->
-    [?f("-record(~p,~n", [Msg]),
+
+    [?f("-ifndef(~s_PB_H).~n", [string:to_upper(atom_to_list(Msg))]),
```

Comment:
This breaks with message names containing dots, such as the record `UninterpretedOption.NamePart` in the descr_src subdir. A clean compile results in

    descr_src/gpb_descriptor.hrl:16: badly formed 'ifndef'

Build the atom in Erlang code and use `~p` instead of `~s` like this (except the line gets too long):

    ?f("-ifndef(~p).~n", [list_to_atom(string_to_upper(atom_to_list(Msg)) ++ "_PB_H")])